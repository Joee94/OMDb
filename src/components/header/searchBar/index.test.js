import * as React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from './index';

describe('SearchBar', () => {
	describe('when not loading', () => {
		const contextValues = { state: { response: { loading: false }, searchValue: 'mock-search-value' } };
		jest.spyOn(React, 'useContext').mockImplementation(() => contextValues);
		const component = renderer.create(<SearchBar />);
		it('should create the SearchBar component', () => {
			expect(component).toMatchSnapshot();
		});
	});
	describe('when loading', () => {
		const contextValues = { state: { response: { loading: true }, searchValue: 'mock-search-value' } };
		const component = renderer.create(<SearchBar />);
		it('should create the SearchBar component', () => {
			expect(component).toMatchSnapshot();
		});
	});
});
