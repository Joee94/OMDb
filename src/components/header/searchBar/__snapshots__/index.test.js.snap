// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar when loading should create the SearchBar component 1`] = `
<form
  className="searchBar hide-submit"
  onSubmit={[Function]}
>
  <label>
    Search OMDb's film database
  </label>
  <input
    disabled="disabled"
    onChange={[Function]}
    placeholder="Star Wars 2: Electric Boogaloo"
    type="search"
  />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;

exports[`SearchBar when not loading should create the SearchBar component 1`] = `
<form
  className="searchBar hide-submit"
  onSubmit={[Function]}
>
  <label>
    Search OMDb's film database
  </label>
  <input
    disabled={null}
    onChange={[Function]}
    placeholder="Star Wars 2: Electric Boogaloo"
    type="search"
  />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
